### PR TITLE
Remove dependency on jscript in IISExpress installers

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes part of https://github.com/aspnet/AspNetCore-Internal/issues/2022. This changes the jscript custom actions to native.

This is targeting the 2.2 branch, but will only be used for VS. Let's align this with the right preview release of VS.

### CODE REVIEW

**Comment by jkotalik** [comment]
@joeloff I mostly want your thoughts on if it's okay for me to depend on MSXML?

**Comment by jkotalik** [comment]
Offline feedback, this looks good. @anurse can you take a quick peek for sanity.

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/ancm_iis_express.wxs [hunk 1]
```diff
@@ -348,66 +348,26 @@
...
-            ]]>
-        </CustomAction>
+                            Property="CA_REMOVE_CONFIGSECTION"
+                            Value="[IISEXPRESS_APPHOST_CONFIG]"/>
+        <CustomAction Id="CA_REMOVE_CONFIGSECTION" BinaryKey="IISCustomActionDll" DllEntry="RemoveConfigSection" Execute="deferred" Return="check" Impersonate="no" />
+
+        <CustomAction Id="CA_REMOVE_CONFIGSECTION_PROPERTY_TMP"
+                Property="CA_REMOVE_CONFIGSECTION_TMP"
+                Value="[IISEXPRESS_APPHOST_CONFIG_TMP]"/>
+        <CustomAction Id="CA_REMOVE_CONFIGSECTION_TMP" BinaryKey="IISCustomActionDll" DllEntry="RemoveConfigSection" Execute="deferred" Return="check" Impersonate="no" />
```
Comment (anurse) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/ancm_iis_express.wxs]: What's with this `_TMP` version?

**Review by analogrelay — APPROVED** [review]

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/ancm_iis_express.wxs (hunk 1 above)
Comment (anurse) [reply src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/ancm_iis_express.wxs]: Talked in person, I get it now ;)

**Comment by jkotalik** [comment]
@Eilon @anurse putting this up for servicing consider.


# Description:
When installing Visual Studio, may customers were reporting install failures due to a dependency on jscript.dll (see https://developercommunity.visualstudio.com/content/problem/167774/packageidmicrosoftancmiisexpressmsipackageactionin-3.html). It isn't recommended to use jscript.dll any longer because it can be blocked by anti-virus and is generally not recommended.

# Customer Impact
The IISExpress msi should become more reliable as it doesn't depend on jscript anylonger.

# Regression?
No regression, this behavior has always existed in the IISExpress installer (as far back as the 15.5 release of ANCM for IISExpress). 

# Risk:
Low. Behavior shouldn't change at all, it just changes a custom action from using jscript to using a native custom action.

**Comment by jkotalik** [comment]
Note, this will have to align with a VS release rather than a dotnet release.

**Comment by Eilon** [comment]
Can you share (internally via email) any telemetry to show how common the install failures are?

**Comment by vivmishra** [comment]
Approved for 2.2.5
Wait for branch to open.

- Loks like it does not impact the hosting bundle. So, only required for VS insertion (ANCM).

**Comment by dougbu** [comment]
Branches are now open in this repo.

However, we've previously listed Microsoft.AspNetCore.AspNetCoreModuleV2 in PatchConfig.props. Suspect that needs to be done this time as well. Once that's done :shipit:

**Comment by dougbu** [comment]
@jkotalik the deadline for merging 2.x fixes is currently the 15th. Please make the necessary changes and get this in soon so that we can start working on official builds a bit early.

**Comment by dougbu** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=151108

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

I've triaged the above build.

**Review by dougbu — APPROVED** [review]

**Pull request merged**